components/dashboard/src/settings/Plans.tsx

```diff
@@ -4,13 +4,461 @@
+        if (!chargebeeClient || !confirmUpgradeToPlan) {
+            return;
+        }
+        try {
+            (event.target as HTMLButtonElement).disabled = true;
+            if (!paidSubscription) {
+                await new Promise((resolve, reject) => {
+                    chargebeeClient.checkout((paymentServer) => paymentServer.checkout(confirmUpgradeToPlan.chargebeeId), {
+                        success: resolve,
+                        error: reject,
+                    });
+                });
+            } else {
+                await server.subscriptionUpgradeTo(paidSubscription.uid, confirmUpgradeToPlan.chargebeeId);
+            }
+            setPendingUpgrade({
+              ... confirmUpgradeToPlan,
+              pendingSince: Date.now(),
+            });
+        } catch (error) {
+            console.error('Upgrade Error', error);
+        } finally {
+            setConfirmUpgradeToPlan(undefined);
+        }
+    }
+    const confirmDowngrade = (to: Plan) => {
+        if (pendingUpgradePlan || pendingDowngradePlan) {
+            // Don't downgrade if we're still waiting for a Chargebee callback
+            return;
+        }
+        if (scheduledDowngradePlanId) {
+            // Don't downgrade if another downgrade is already scheduled (first that one should happen or be be cancelled)
+            return;
+        }
+        if ((paidSubscription?.paymentReference || '').startsWith("github:")) {
+            const url = (gitHubUpgradeUrls || []).find(u => u.planID == to.githubId);
+            if (!url) {
+                return;
+            }
+            window.location.href = url.url;
+            return;
+        }
+        setConfirmDowngradeToPlan(to);
+    }
+    const doDowngrade = async (event: React.MouseEvent) => {
+        if (!confirmDowngradeToPlan || !paidSubscription) {
+            return;
+        }
+        try {
+            (event.target as HTMLButtonElement).disabled = true;
+            if (Plans.isFreePlan(confirmDowngradeToPlan.chargebeeId)) {
+                await server.subscriptionCancel(paidSubscription.uid);
+            } else {
+                await server.subscriptionDowngradeTo(paidSubscription.uid, confirmDowngradeToPlan.chargebeeId);
+            }
+            setPendingDowngrade({
+              ... confirmDowngradeToPlan,
+              pendingSince: Date.now(),
+            });
+        } catch (error) {
+            console.error('Downgrade Error', error);
+        } finally {
+            setConfirmDowngradeToPlan(undefined);
+        }
+    }
+
+    const planCards = [];
+
+    // Plan card: Free a.k.a. Open Source (or Professional Open Source)
+    const openSourceFeatures = <>
+        <p className="truncate">✓ Public Repositories</p>
+        <p className="truncate">✓ 4 Parallel Workspaces</p>
+        <p className="truncate">✓ 30 min Timeout</p>
+    </>;
+    if (currentPlan.chargebeeId === freePlan.chargebeeId) {
```

Comment:
Actually, I'm not sure I fully understand the suggestion. To me it was about loading `chargebee.js` as late as possible (which means everything unrelated to payment is faster, but the initial load of payment-related things is slower). Are we instead talking about a way to make the payment-related things load faster? (Happy to discuss in a call)

Reply:
Checking this in action again, page load seems good. State loading seems slow but this could be expected. [Call me maybe](https://youtu.be/fWNaR-rxAic) later or tomorrow! ❤️